packages/framework-types/src/searcher.ts

```diff
@@ -1,8 +1,15 @@
 /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/indent */
 import { UUID } from './concepts'
+import { ReadModelResult } from './envelope'
 import { Class } from './typelevel'
 
-export type SearcherFunction<TObject> = (className: string, filters: FilterFor<TObject>) => Promise<Array<any>>
+export type SearcherFunction<TObject> = (
+  className: string,
+  filters: FilterFor<TObject>,
+  limit?: number,
+  afterCursor?: any,
+  paginatedVersion?: boolean
```

Comment:
Would you mind creating a ticket to remove the `paginatedVersion` related code in the next version?

Reply:
Sorry, I missed this comment. Is on #888